core/sdk-core/src/main/java/software/amazon/awssdk/core/exception/SdkServiceException.java

```diff
@@ -76,21 +76,30 @@ public int statusCode() {
     }
 
     /**
-     * Specifies whether or not an exception may have been caused by clock skew.
+     * Specifies whether an exception may have been caused by clock skew.
      */
     public boolean isClockSkewException() {
         return false;
     }
 
     /**
-     * Specifies whether or not an exception is caused by throttling.
+     * Specifies whether an exception is caused by throttling.
      *
      * @return true if the status code is 429, otherwise false.
      */
     public boolean isThrottlingException() {
         return statusCode == HttpStatusCode.THROTTLING;
     }
 
+    /**
+     * Specifies whether an exception is retryable.
+     *
+     * @return true if the exception is classified as retryable.
+     */
+    public boolean isRetryableException() {
+        return false;
```

Comment:
This document is misleading, it suggest that all types extending `SdkServiceException`  can be retried but that's not the case, we have tests to validate otherwise, e.g., [here](https://github.com/aws/aws-sdk-java-v2/blob/377f2a289cf8eeb66305424a61805b9532446593/core/aws-core/src/test/java/software/amazon/awssdk/awscore/retry/AwsRetryPolicyTest.java#L100) (and below)

```java
    @Test
    public void doesNotRetryOnNonRetryableErrorCode() {
        assertFalse(shouldRetry(applyErrorCode("ValidationError")));
    }
```

Notice that the test [is setup](https://github.com/aws/aws-sdk-java-v2/blob/377f2a289cf8eeb66305424a61805b9532446593/core/aws-core/src/test/java/software/amazon/awssdk/awscore/retry/AwsRetryPolicyTest.java#L119) using a `AwsServiceException` *and* `AwsServiceException` [extends](https://github.com/aws/aws-sdk-java-v2/blob/f14f4a29671b5c603d2c5899069d13e97cad3cae/core/aws-core/src/main/java/software/amazon/awssdk/awscore/exception/AwsServiceException.java#L43) `SdkServiceException`.

I will rephrase it in a different PR to make make this clear.

Reply:
I see, thanks for the context. So it is OKAY to have `isRetryableException()` by default be false here? It is indeed quite misleading and confusing to look at the documentaion which says it is retryable by default, and then seeing this method returns false.

Reply:
Yeah, you're right, I will update the javadocs to clarify this point.

Reply:
> So it is OKAY to have isRetryableException() by default be false here? 

Yes, that's actually the default right now. Notice that we test several conditions to determine if an exception is retryable and we test them all using  "OR" semantics and consider an exception to be "retryable" if **any** returns `true`. E.g., while this part might be false and thus fail this test the return status might be 500 which will return true, see [this setup](https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/retry/SdkDefaultRetrySetting.java#L73).

I will update the javadocs though to make it clear that this method will return true if *the modeled* exception is marked as retryable. This is probably part of the confusion.